app/celery/provider_tasks.py

```diff
@@ -10,7 +14,24 @@
 from app.dao.notifications_dao import update_notification_status_by_id
 from app.delivery import send_to_providers
 from app.exceptions import NotificationTechnicalFailureException
-from app.models import NOTIFICATION_TECHNICAL_FAILURE
+from app.models import (
+    NOTIFICATION_FAILED,
+    NOTIFICATION_SENT,
+    NOTIFICATION_TECHNICAL_FAILURE,
+)
+
+
+@notify_celery.task(bind=True, name="check_sms_delivery_receipt", max_retries=3, default_retry_delay=300)
+def check_sms_delivery_receipt(self, message_id, notification_id):
+    current_app.logger.warning(f"CHECKING DELIVERY RECEIPT for {message_id} {notification_id}")
+    cloudwatch_client = AwsCloudwatchClient()
+    cloudwatch_client.init_app(current_app)
+    status, provider_response = cloudwatch_client.check_sms(message_id, notification_id)
+    if status == 'success':
+        status = NOTIFICATION_SENT
+    else:
```

Comment:
[question] Are you certain there will only be a log entry once the result is either success or failed? There's no pending state it could be in? Or retrying?

Reply:
Ah, that's a good question.  I will have to poke around to see.  I guess we will have to have enough of a delay that retrying time is up.